sqlglot/dialects/athena.py

```diff
@@ -1,31 +1,131 @@
 from __future__ import annotations
 
+import typing as t
+
 from sqlglot import exp
 from sqlglot.dialects.trino import Trino
-from sqlglot.tokens import TokenType
+from sqlglot.dialects.hive import Hive
+from sqlglot.tokens import Token, TokenType
+
+
+def _parse_as_hive(raw_tokens: t.List[Token]) -> bool:
+    if len(raw_tokens) > 0:
+        first_token = raw_tokens[0]
+        if first_token.token_type == TokenType.CREATE:
```

Comment:
I'm not sure if this approach is robust, for example what if you have leading CTEs? The first token would then be `WITH`. You could also have redundant semi-colons, etc.

Reply:
Yeah I was anticipating needing to tweak this. I forgot about CTE's, i'll add support for that.

How would you deal with redundant semicolons? Would you first filter the token stream to just the ones we care about and then do the checks?

Reply:
So if you have leading CTE's, the existing logic works and returns `False`. It doesnt need to check for them because only a `SELECT` query would have leading CTE's. Returning `False` triggers the use of the Trino tokenizer.

If you have a `CREATE TABLE .. AS WITH (...) SELECT` query, a `SELECT` still appears in the tokens so this is still correctly detected as a CTAS and returns `False` which triggers the use of the Trino tokenizer

Reply:
How do you test redundant semicolons?

Trying to parse something like `; CREATE SCHEMA FOO;` just throws a parse error

Reply:
> So if you have leading CTE's, the existing logic works and returns False. It doesnt need to check for them because only a SELECT query would have leading CTE's. All SELECT queries should use the Trino tokenizer, so returning False triggers the use of the Trino tokenizer.

Ah, interesting, I didn't realize that - that simplifies the problem then.

> How do you test redundant semicolons?

Using `parse`:

```python
>>> import sqlglot
>>> sqlglot.parse("; create schema foo ;;")
[None, Create(
  this=Table(
    db=Identifier(this=foo, quoted=False)),
  kind=SCHEMA), None]
```

Tbh this is an edge case but good to handle anyway, you can skip the leading tokens until you find the first non-semicolon or something.

Reply:
I struggled to write a test for this and eventually realized that the issues I had that I _thought_ were parsing issues were really tokenization issues. It appears the Trino parser is capable of handling all the Hive DDL as long as it is tokenized correctly.

So I was able to remove the delegation to the Hive parser entirely and just worry about the generation side. I may need to revisit this in future but I wasn't able to find a query to add to the tests that uses Hive syntax and also causes the Trino parser to fail